semgrep-core/tests/rules/taint_propagator_lambda.java

```diff
@@ -0,0 +1,31 @@
+class Test {
+
+    public void func1() {
+        String nodePred = request.getParameter("h");
+        String nodeSucc = nodePred.toString();
+        //ruleid: test
+        Runtime.getRuntime().exec(nodeSucc);
+    }
+
+    public void func2() {
+        String nodePred = request.getParameter("h");
+        String nodeSucc = fnTest(nodePred);
+        //ruleid: test
+        Runtime.getRuntime().exec(nodeSucc);
+    }
+
+    public void func3() {
+        String nodePred = request.getParameter("h");
+        NodeSucc nodeSucc = new NodeSucc(nodePred);
+        //ruleid: test
+        Runtime.getRuntime().exec(nodeSucc);
+    }
+
+    public void func4() {
```

Comment:
This feels like it should be the only tests.
What are testing the other func1/func2/func3?
I don't even see the difference between their code.

Reply:
That was the example reported by the user. For debugging, I always minimize the example. But for testing, I tend to include it as-is, with the hope to increase coverage.